moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame.cpp

```diff
@@ -211,24 +211,36 @@ void MotionPlanningFrame::setItemSelectionInList(const std::string& item_name, b
 
 void MotionPlanningFrame::allowExternalProgramCommunication(bool enable)
 {
+  if (first_time_)
+  {
+    return;
+  }
```

Comment:
why is this needed here?

At first glance this seems unnecessary.

Reply:
This is required because when resuming saved UI state (before `first_time_ == true`), this function will be executed because of the Qt signal, while something required for this doesn't complete loading (I guess), then it will segfault indefinitely. The quick fix applied here:

1. Don't execute unless `first_time_` load completes;
2. [Manually trigger the function after `first_time_` load](https://github.com/ros-planning/moveit/pull/823/files#diff-0b8bf166073217818c9c79b974152957R346).

Reply:
Not a neat solution, as you say.
It would be interesting for me to know what goes wrong here. The only thing I can imagine is that `planning_display_->getRobotInteraction()` is still a nullptr.

Please add a proper comment explaining this reasoning above the condition.

Reply:
The `marker_access_lock_` in the interactive marker is the issue. I cannot think of a nicer fix for now :disappointed: 

> Thread 1 "rviz" received signal SIGSEGV, Segmentation fault.
__GI___pthread_mutex_lock (mutex=0x158) at ../nptl/pthread_mutex_lock.c:67
> `#0`  __GI___pthread_mutex_lock (mutex=0x158) at ../nptl/pthread_mutex_lock.c:67
> `#1`  0x00007ffff7a8b098 in boost::unique_lock<boost::mutex>::lock() () from /opt/ros/kinetic/lib/librviz.so
> `#2`  0x00007fff90d87a55 in robot_interaction::RobotInteraction::toggleMoveInteractiveMarkerTopic(bool) ()
   from /home/phy/moveit-clone/devel/lib/libmoveit_robot_interaction.so.0.9.11
> `#3`  0x00007fff91ececc6 in moveit_rviz_plugin::MotionPlanningFrame::allowExternalProgramCommunication(bool) ()
   from /home/phy/moveit-clone/devel/lib/libmoveit_motion_planning_rviz_plugin_core.so.0.9.11
> `#4`  0x00007ffff1f80d2a in QMetaObject::activate(QObject*, int, int, void**) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
> `#5`  0x00007ffff77fe312 in QAbstractButton::toggled(bool) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
> `#6`  0x00007ffff7561645 in ?? () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
> `#7`  0x00007ffff7562acc in QAbstractButton::setChecked(bool) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
> `#8`  0x00007fff91f4113c in moveit_rviz_plugin::MotionPlanningDisplay::load(rviz::Config const&) ()

Reply:
... which means `getRobotInteraction()` is not a valid object yet, I guess.
Fair enough. Please add a comment that explains this reasoning above the `if` here as well as in the lines below that trigger the lazy load.